src/inmanta/deploy/scheduler.py

```diff
@@ -675,37 +713,89 @@ async def report_resource_state(
             if deployment_result is not None:
                 # first update state, then send out events
                 self._deploying_latest.remove(resource)
+
+                previous_deployment_result = state.deployment_result
+
                 state.deployment_result = deployment_result
                 self._work.finished_deploy(resource)
+
+                # Set of dependant resources that are interested in deploys of the current resource
+                concerned_resources: set[ResourceIdStr] = set()
+                provides: Set[ResourceIdStr] = self._state.requires.provides_view().get(resource, set())
+
                 if deployment_result is DeploymentResult.DEPLOYED:
                     self._state.dirty.discard(resource)
+                    is_dirty = False
+
+                    if previous_deployment_result != DeploymentResult.DEPLOYED:
+                        # This resource went from not deployed to deployed
+                        # we have to inform its dependants regardless of event propagation
+                        # so that we can try to re-deploy them in case this resource's previous
+                        # failure was the reason they were skipped
+
+                        # FIXME: stop informing resource skipped because of
+                        #  a custom handler skip via this mechanism
+
+                        dependant_resources: set[ResourceIdStr] = set()
```

Comment:
I would find it easier to understand if you would update `concerned_resources` immediately instead of first building up `dependant_resources` and then copy it into `concerned_resources`